pyzo/core/shellStack.py

```diff
@@ -725,21 +726,49 @@ def useFound(self):
         # Set newfound interpreter
         if self._the_exe:
             configs = pyzo.config.shellConfigs2
-            if not configs:
+            config_to_use = None
+            # if there is no shell : create one and use it
+            # if auto_useFound == 2 and there is an autodetected shell : update it
+            # if auto_useFound == 2 and there is no autodetected shell : create one and use it
+            if pyzo.config.settings.auto_useFound == 2:
+                for conf in configs:
+                    if getattr(conf, "wasAutodetected", False):
+                        config_to_use = conf
+                        break
+            if (
+                len(configs) == 0
+                or pyzo.config.settings.auto_useFound == 2
+                and config_to_use is None
+            ):
```

Comment:
This can be `if config_to_use is None`, right? (and then the `if` below can be an `else`)

Reply:
No. Imagine there are one or several configs but none if them was autogenerated, and we are not in mode 2 : we should not generate a new config, but ``config_to_use`` will be None.

Reply:
Right, my bad.